src/blob_file_iterator.h

```diff
@@ -58,7 +58,10 @@ class BlobFileIterator {
   Status status_;
   bool valid_{false};
 
-  BlobDecoder decoder_;
+  std::unique_ptr<UncompressionDict> uncompression_dict_;
+  std::unique_ptr<BlobDecoder> decoder_ =
+      std::unique_ptr<BlobDecoder>(new BlobDecoder);
```

Comment:
remind me why we need to make this a pointer?

Reply:
The decoder has to reconstruct at my first implementaion, so it needed a smart pointer. But, I've done some refactoring in last few commits, the decoder here doesn't really need a smart pointer, I'll change it! Nice catch!